src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -139,6 +139,17 @@ def attach_data(
             predict_dataloaders=predict_dataloaders,
         )
         self.attach_datamodule(model, datamodule=datamodule)
+
+        # Validate that the required data sources are available
+        if self.trainer.state.fn == TrainerFn.FITTING:
+            _check_dataloader_none(train_dataloaders, self._train_dataloader_source, self.trainer.state.fn)
```

Comment:
This should check validation too

Reply:
Why though. We only want to check what is required. The val dataloader is optional in fitting. If val dataloader is not provided, and val step is implemented, we skip validation altogether, with a warning:

> You defined a `validation_step` but have no `val_dataloader`. Skipping val loop

I'm pretty sure this should be kept. Can you please check again.